client/autoclient.go

```diff
@@ -0,0 +1,452 @@
+package client
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"sync"
+
+	"decred.org/dcrwallet/v3/errors"
+	"decred.org/dcrwallet/v3/wallet"
+	"decred.org/dcrwallet/v3/wallet/udb"
+	"github.com/decred/dcrd/chaincfg/chainhash"
+	"github.com/decred/dcrd/chaincfg/v3"
+	"github.com/decred/dcrd/dcrutil/v4"
+	"github.com/decred/dcrd/txscript/v4"
+	"github.com/decred/dcrd/txscript/v4/stdaddr"
+	"github.com/decred/dcrd/wire"
+	"github.com/decred/slog"
+)
+
+type DialFunc func(ctx context.Context, network, addr string) (net.Conn, error)
+
+type Policy struct {
+	MaxFee     dcrutil.Amount
+	ChangeAcct uint32 // to derive fee addresses
+	FeeAcct    uint32 // to pay fees from, if inputs are not provided to Process
+}
+
```

Comment:
Maybe it is best to insure that dcrwallet can satisfy this interface with the following somewhere:

`var _ WalletFetcher = (*wallet.Wallet)(nil)`

That way if a method changes in dcrwallet you will know when building here that this interface is no longer compatible.